src/app/auth/components/registration/registration.component.ts

```diff
@@ -1,22 +1,44 @@
-import { Component, Output, EventEmitter } from '@angular/core';
+import { Component, Output, EventEmitter, ChangeDetectionStrategy } from '@angular/core';
+import { FormGroup, FormBuilder, Validators, AbstractControl } from '@angular/forms';
 
+import { CustomValidationService } from '../../services/custom-validation.service';
 import { IRegistration } from '../../interfaces/registration.interface';
 
-
 @Component({
   selector: 'app-registration',
   templateUrl: './registration.component.html',
   styleUrls: ['./registration.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class RegistrationComponent {
 
   @Output()
-  public formSubmit = new EventEmitter();
+  public formSubmit = new EventEmitter<IRegistration>();
+
+  public readonly registrationForm!: FormGroup;
 
-  public formControls: Partial<IRegistration> = {};
+  public get confirmPasswordControl(): AbstractControl {
+    return this.registrationForm.controls['confirmPassword'];
```

Comment:
You forgot to improve this solution. Try to use the following code for getting AbstractControl:
`return this.registrationForm.get('confirmPassword');`